### PR TITLE
FIX React components for reactstrap 8 upgrade

### PR DESCRIPTION
**Pull request by emteknetnz**
Fixes for https://github.com/silverstripe/silverstripe-admin/issues/1126

### CODE REVIEW

**Review by Cheddam — CHANGES_REQUESTED** [review]
Good to get some basic coverage in place on these components. Some incorrect names, and a couple of places I think we could be a _little_ more specific about the output (I'm aware that these are 'MVP' tests, but we're unlikely to touch them again anytime soon.)

File: client/src/components/HtmlReadonlyField/HtmlReadonlyField.js [hunk 1]
```diff
@@ -36,12 +21,7 @@ class HtmlReadonlyField extends Component {
 
   render() {
     return (
-      <Input
-        plaintext
-        {...this.getInputProps()}
-        // eslint-disable-next-line react/no-danger
-        dangerouslySetInnerHTML={this.getContent()}
-      />
+      <Input plaintext {...this.getInputProps()} tag="p">{this.props.value}</Input>
```
[the remark this answers is not in the document]
Comment (emteknetnz) [reply client/src/components/HtmlReadonlyField/HtmlReadonlyField.js:24]: I've added an additional case to the storybook that includes an HTML value, it'll render like this

image

File: client/src/components/ResizeAware/tests/ResizeAware-test.js [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+      errorSpy.mockClear();
+      warnSpy.mockClear();
+    });
+
+    it('renders with children', () => {
+      const reactWrapper = mount(
+        <ResizeAware {...props} />
+      );
+
+      expect(reactWrapper.find('div')).toHaveLength(1);
```
[the remark this answers is not in the document]
Comment (emteknetnz) [reply client/src/components/ResizeAware/tests/ResizeAware-test.js:30]: Note really https://github.com/silverstripe/silverstripe-admin/blob/1/client/src/components/ResizeAware/ResizeAware.js

File: client/src/components/Toolbar/tests/Toolbar-test.js [hunk 3]
```diff
@@ -0,0 +1,46 @@
...
+      expect(warnSpy).not.toHaveBeenCalled();
+    });
+
+    it('renders with a back button', () => {
+      const props2 = { ...props, showBackButton: true };
+      const reactWrapper = mount(
+        <Toolbar {...props2} />
+      );
+
+      expect(reactWrapper.find('div.toolbar')).toHaveLength(1);
```
[the remark this answers is not in the document]
Comment (emteknetnz) [reply client/src/components/Toolbar/tests/Toolbar-test.js:41]: Couldn't find a reliable way to do this, so I removed the back button unit test and created an MVP story book that includes a back button 'test'

image

File: client/src/components/MobileMenuToggle/tests/MobileMenuToggle-test.js [hunk 4]
```diff
@@ -0,0 +1,47 @@
...
+      expect(warnSpy).not.toHaveBeenCalled();
+    });
+
+    it('renders open', () => {
+      const props2 = { ...props, isOpen: true };
+      const reactWrapper = mount(
+        <MobileMenuToggle {...props2} />
+      );
+
+      expect(reactWrapper.find('button')).toHaveLength(1);
```
[the remark this answers is not in the document]
Comment (emteknetnz) [reply client/src/components/MobileMenuToggle/tests/MobileMenuToggle-test.js]: Updated

File: client/src/components/SingleSelectField/tests/SingleSelectField-test.js [hunk 5]
```diff
@@ -0,0 +1,76 @@
...
+  value: 'My value',
+  readOnly: false,
+  disabled: false,
+  source: [],
+  data: {
+    emptyString: 'Any'
+  },
+};
+
+describe('HiddenField', () => {
```
[the remark this answers is not in the document]
Comment (emteknetnz) [reply client/src/components/SingleSelectField/tests/SingleSelectField-test.js]: Updated

File: client/src/components/Focusedzone/tests/Focusedzone-test.js [hunk 6]
```diff
@@ -0,0 +1,38 @@
...
+
+const errorSpy = jest.spyOn(global.console, 'error');
+const warnSpy = jest.spyOn(global.console, 'warn');
+
+const props = {
+  className: 'my-classname',
+  onClickOut: () => {}
+};
+
+describe('FieldGroup', () => {
```
[the remark this answers is not in the document]
Comment (emteknetnz) [reply client/src/components/Focusedzone/tests/Focusedzone-test.js]: Updated

File: client/src/components/HtmlEditorField/tests/HtmlEditorField-test.js [hunk 7]
```diff
@@ -0,0 +1,53 @@
...
+  data: {
+    editorjs: 'I AM AN EDITOR_JS',
+    attributes: {
+      'data-abc': '123',
+      'data-def': '456',
+    }
+  }
+};
+
+describe('FieldGroup', () => {
```
[the remark this answers is not in the document]
Comment (emteknetnz) [reply client/src/components/HtmlEditorField/tests/HtmlEditorField-test.js]: Updated

File: client/src/components/Preview/tests/Preview-test.js [hunk 8]
```diff
@@ -0,0 +1,39 @@
...
+const props = {
+  className: 'my-classname',
+  itemLinks: {},
+  itemId: 123,
+  onBack: () => {},
+  moreActions: [],
+  ViewModeComponent: () => <div />,
+};
+
+describe('FieldGroup', () => {
```
[the remark this answers is not in the document]
Comment (emteknetnz) [reply client/src/components/Preview/tests/Preview-test.js]: Updated

File: client/src/components/NotFoundComponent/tests/NotFoundComponent-test.js [hunk 9]
```diff
@@ -0,0 +1,36 @@
...
+const errorSpy = jest.spyOn(global.console, 'error');
+const warnSpy = jest.spyOn(global.console, 'warn');
+
+const props = {
+  itemName: 'MyItemName',
+  name: 'MyName',
+  value: 'My value'
+};
+
+describe('FieldGroup', () => {
```
[the remark this answers is not in the document]
Comment (emteknetnz) [reply client/src/components/NotFoundComponent/tests/NotFoundComponent-test.js]: Updated

File: client/src/components/MobileMenuToggle/tests/MobileMenuToggle-test.js [hunk 10]
```diff
@@ -0,0 +1,47 @@
...
+const errorSpy = jest.spyOn(global.console, 'error');
+const warnSpy = jest.spyOn(global.console, 'warn');
+
+const props = {
+  isOpen: false,
+  onClick: () => {},
+  controls: 'something',
+};
+
+describe('FieldGroup', () => {
```
[the remark this answers is not in the document]
Comment (emteknetnz) [reply client/src/components/MobileMenuToggle/tests/MobileMenuToggle-test.js]: Updated

File: client/src/components/LiteralField/tests/LiteralField-test.js [hunk 11]
```diff
@@ -0,0 +1,38 @@
...
+
+const props = {
+  id: 'my-id',
+  name: 'MyName',
+  className: 'my-classname',
+  extraClass: 'my-extra-class',
+  value: '<h2>My literal heading</h2><p>My literal content</p>',
+};
+
+describe('FieldGroup', () => {
```
[the remark this answers is not in the document]
Comment (emteknetnz) [reply client/src/components/LiteralField/tests/LiteralField-test.js]: Updated

File: client/src/components/LabelField/tests/LabelField-test.js [hunk 12]
```diff
@@ -0,0 +1,40 @@
...
+  id: 123,
+  className: 'my-classname',
+  extraClass: 'my-extra-class',
+  title: 'my-title',
+  data: {
+    target: 'my-target'
+  }
+};
+
+describe('FieldGroup', () => {
```
[the remark this answers is not in the document]
Comment (emteknetnz) [reply client/src/components/LabelField/tests/LabelField-test.js]: Updated

File: client/src/components/HtmlReadonlyField/HtmlReadonlyField.js (hunk 1 above)
[the remark this answers is not in the document]
Comment (Cheddam) [reply client/src/components/HtmlReadonlyField/HtmlReadonlyField.js:24]: That confirms this is a breaking change. Was `dangerouslySetInnerHTML={this.getContent()}` not working? You might need to nest it in a child element if so.

[the remark this answers is not in the document]
Comment (emteknetnz) [reply client/src/components/HtmlReadonlyField/HtmlReadonlyField.js:24]: >> dangerouslySetInnerHTML={this.getContent()} not working? 

Yeah there was a big red error thrown

Screenshot above refers to this newly added story:

https://github.com/silverstripe/silverstripe-admin/pull/1128/files#diff-13544a97c2d6bb8894bbff043a685e98R6

I believe this is how it's supposed to work?

[the remark this answers is not in the document]
Comment (Cheddam) [reply client/src/components/HtmlReadonlyField/HtmlReadonlyField.js:24]: That's not how it worked before, though - this change would be similar to escaping HTML in field descriptions. If this is no longer directly supported by the `Input` component, I recommend taking the approach I suggested of nesting the content in a child element, e.g.

```suggestion
      <Input plaintext {...this.getInputProps()} tag="p"><div dangerouslySetInnerHTML={this.getContent()} /></Input>
```

[the remark this answers is not in the document]
Comment (emteknetnz) [reply client/src/components/HtmlReadonlyField/HtmlReadonlyField.js]: Have reverted behaviour. `dangerouslySetInnerHTML` with reactstrap8 as long as `tag="p"` is present

**Review by Cheddam — APPROVED** [review]

**Pull request merged**